test/Data/StoreSpec.hs

```diff
@@ -31,7 +31,9 @@ import           Data.IntSet (IntSet)
 import qualified Data.List.NonEmpty as NE
 import           Data.Map (Map)
 import           Data.Monoid
+#if !MIN_VERSION_primitive(0,7,0)
```

Comment:
This is an unrelated change to fix the build against newer versions of `primitive`. It is still not sufficient to get the test suite working on recent GHCs and recent dependencies (I tried 8.6.5) so to test the instance, I’ve simply uncommented everything but `Integer` in the “Store on all monomorphic instances” tests.